src/symbolize/gimli/elf.rs

```diff
@@ -345,25 +347,17 @@ fn locate_build_id(build_id: &[u8]) -> Option<PathBuf> {
     }
 
     let mut path =
-        Vec::with_capacity(BUILD_ID_PATH.len() + BUILD_ID_SUFFIX.len() + build_id.len() * 2 + 1);
-    path.extend(BUILD_ID_PATH);
-    path.push(hex(build_id[0] >> 4));
-    path.push(hex(build_id[0] & 0xf));
-    path.push(b'/');
+        String::with_capacity(BUILD_ID_PATH.len() + BUILD_ID_SUFFIX.len() + build_id.len() * 2 + 1);
+    path.push_str(BUILD_ID_PATH);
+    path.push(char::from_digit((build_id[0] >> 4) as u32, 16)?);
+    path.push(char::from_digit((build_id[0] & 0xf) as u32, 16)?);
```

Comment:
Hmm, this is significantly noisier in practice... I think I'd prefer to keep `fn hex`.